src/block/RedMushroom.php

```diff
@@ -43,10 +43,12 @@ public function onNearbyBlockChange() : void{
 
 	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
 		$down = $this->getSide(Facing::DOWN);
-		if(!$down->isTransparent()){
+		$position = $this->getPosition();
+		$lightLevel = $position->getWorld()->getFullLightAt($position->x, $position->y, $position->z);
+		//TODO: nylium support
+		if(($lightLevel <= 12 && !$down->isTransparent()) || $down->isSameType(VanillaBlocks::MYCELIUM()) || $down->isSameType(VanillaBlocks::PODZOL())){
```

Comment:
ID comparisons should be used here. VanillaBlocks creates a new object for every call.

You may want to retarget this to next-major.

Reply:
For the retarget, I don't see much use in the sense that it was in the logical continuation of #4886 what concerned the placement of the blocks.

Reply:
Fine. Nonetheless, IDs should be used for comparison, not VanillaBlocks.

Reply:
Alright